python-frontend/src/main/java/org/sonar/python/types/InferredTypes.java

```diff
@@ -399,6 +417,9 @@ static boolean isTypeClassCompatibleWith(Symbol typeClass, InferredType other) {
     if (other instanceof UnionType) {
       return ((UnionType) other).types().stream().anyMatch(t -> InferredTypes.isTypeClassCompatibleWith(typeClass, t));
     }
+    if (other instanceof UnknownClassType) {
+      return InferredTypes.areSymbolsCompatible(typeClass, ((UnknownClassType) other).typeClass());
```

Comment:
Do we need this?
Given the type is unknown, I feel we should be conservative here and return true like with `AnyType`.
Otherwise we'd risk some false positives I guess?